types/react-dynamic-number/index.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for react-dynamic-number 1.7
+// Project: https://github.com/uhlryk/react-dynamic-number
+// Definitions by: Eugene Rodin <https://github.com/eugrdn>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+import * as React from 'react';
+
+/**
+ * remove Diff & Omit when in will be placed in TS from scratch
+ */
+type Diff<T extends string, U extends string> = ({[P in T]: P} & {[P in U]: never} & {[x: string]: never})[T];
+type Omit<T, K extends keyof T> = {[P in Diff<keyof T, K>]: T[P]};
+
+interface OmittedFields {
+  ref: any;
+  value: any;
+  onChange: any;
+  placeholder: any;
+}
+
+type BaseInputProps = Partial<
+  Omit<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, keyof OmittedFields>
+>;
+
+type Value = number | '';
+type Separator = '.' | ',';
+type Thousand = boolean | ' ';
+
+export interface DynamicNumberProps extends BaseInputProps {
+  value?: Value;
+  separator?: Separator;
+  thousand?: Thousand;
+  integer?: number;
+  fraction?: number;
+  positive?: boolean;
+  negative?: boolean;
+  placeholder?: string;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>, modelValue: number, viewValue: string) => void;
+}
+
+declare class DynamicNumber extends React.Component<DynamicNumberProps> {}
+export default DynamicNumber;
```

Comment:
`export default class DynamicNumber extends React.Component<DynamicNumberProps> {}` should work.

Reply:
Is that a common code style requirement of the DefinitelyTyped or just your opinion?

Reply:
We don't have a lint rule for it, but I think we would if we had the time. No need to declare something and then export it if you can export directly.

Reply:
Ok. So, if I remove the `declare` statement there are some tslint errors appearing for the types that not needed to be exported (like Omit or BaseInputProps): "Prefer to explicitly write 'export' for an external module export."

Reply:
Just change that to `export type Diff = ...`. A declaration in a `.d.ts` module declaration is always exported whether you write `export` or not, so the lint rule is just asking to make it explicit.